### PR TITLE
Add versioning

### PR DESCRIPTION
**Pull request by lauraghiorghisor-tw**
Introduce versioning for the package:
- add a CHANGELOG
- update README to include instructions for publishing to NPM

[Trello](https://trello.com/c/PhgX6wzC/2540-distribute-the-govspeak-visual-editor-via-node-package-manager)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,3 +31,36 @@ jobs:
 
   vitest:
     uses: ./.github/workflows/vitest.yml
+
+  release:
+    needs: vitest
```
Comment (lauraghiorghisor-tw) [inline .github/workflows/ci.yml:36]: I was thinking to make it depend on all the above jobs, but I should hope that you cannot merge in if they don't pass anyway, so depending on test only should be ok?

Comment (ChrisBAshton) [reply .github/workflows/ci.yml:36]: Sure 👍 that's also consistent with how it's handled in other places, e.g. [in govuk_publishing_components](https://github.com/alphagov/govuk_publishing_components/blob/main/.github/workflows/ci.yml)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -31,3 +31,36 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+          cache: npm
+          registry-url: "https://registry.npmjs.org"
+      - run: npm ci
+      - run: npm run build
+      - name: Establish version
```
Comment (ChrisBAshton) [inline .github/workflows/ci.yml:50]: Is it worth adding to the commit message that this code is copied from https://github.com/alphagov/paste-html-to-govspeak/blob/975879e4630d90625f958f4993b9508014825db1/.github/workflows/ci.yml#L54-L70?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍

**Pull request merged**